### DISCUSSION TITLE
connectionTimeout Significance for JedisSentinelPool()

### DISCUSSION THREAD

**Discussion by anil181** [xetorthio/jedis#2620] [Q&A] [answered]
If i want to set infinite timeout for the connectionTimout, should i set the value to 0? 
Is setting it to 0 a good idea or it should be some finite number of milliseconds?
 
https://javadoc.io/doc/redis.clients/jedis/3.6.0/redis/clients/jedis/JedisSentinelPool.html#connectionTimeout

**Comment by yangbodong22011** [comment, marked as answer]
> If i want to set infinite timeout for the connectionTimout, should i set the value to 0?

true

> https://javadoc.io/doc/redis.clients/jedis/3.6.0/redis/clients/jedis/JedisSentinelPool.html#connectionTimeout

in [latest code](https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/JedisSentinelPool.java#L138), infiniteSoTimeout default value is 0.

**Comment by sazzad16** [comment]
connectionTimout is passed directly to Socket constructor. There isn't any check about it in Jedis. So support of setting 0 as is completely dependent on the implementation of Java Socket.
